lib/logger/lib/logger.ex

```diff
@@ -817,20 +833,25 @@ defmodule Logger do
   end
 
   defp macro_log(level, data, metadata, caller) do
-    %{module: module, function: fun, file: file, line: line} = caller
+    %{module: module, function: {fun, arity}, file: file, line: line} = caller
 
     caller =
-      compile_time_application_and_file(file) ++
-        [module: module, function: form_fa(fun), line: line]
+      file
+      |> compile_time_application_and_file()
+      |> Keyword.merge(mfa: {module, fun, arity}, line: line)
 
     {compile_metadata, quoted_metadata} =
       if Keyword.keyword?(metadata) do
         metadata = Keyword.merge(caller, metadata)
-        {metadata, metadata}
+        quoted = Keyword.update!(metadata, :mfa, &Macro.escape/1)
```

Comment:
I think you can do:

    quoted = Macro.escape(Map.new(metadata))

And therefore `quote`+`unquote_splicing` below won't be necessary.

Reply:
I have tried that earlier but it failed with escaping a little bit too much:

```
:gen_event handler Logger.Backends.Console installed in Logger terminating
** (exit) an exception was raised:
    ** (Protocol.UndefinedError) protocol String.Chars not implemented for {:__aliases__, [counter: {LoggerTest, 173}, line: 124], [:CustomTest]} of type Tuple
        (elixir 1.10.0-dev) lib/string/chars.ex:3: String.Chars.impl_for!/1
        (elixir 1.10.0-dev) lib/string/chars.ex:22: String.Chars.to_string/1
        (logger 1.10.0-dev) lib/logger/formatter.ex:176: Logger.Formatter.metadata/1
        (logger 1.10.0-dev) lib/logger/formatter.ex:152: anonymous fn/6 in Logger.Formatter.format/5
        (elixir 1.10.0-dev) lib/enum.ex:2063: Enum."-reduce/3-lists^foldl/2-0-"/3
        (logger 1.10.0-dev) lib/logger/formatter.ex:151: Logger.Formatter.format/5
        (logger 1.10.0-dev) lib/logger/backends/console.ex:186: Logger.Backends.Console.format_event/5
        (logger 1.10.0-dev) lib/logger/backends/console.ex:140: Logger.Backends.Console.log_event/5
        (logger 1.10.0-dev) lib/logger/backends/console.ex:45: Logger.Backends.Console.handle_event/2
        (stdlib 3.9.2) gen_event.erl:577: :gen_event.server_update/4
        (stdlib 3.9.2) gen_event.erl:559: :gen_event.server_notify/4
        (stdlib 3.9.2) gen_event.erl:300: :gen_event.handle_msg/6
        (stdlib 3.9.2) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
```

Reply:
Thanks!